backends/platform/ios7/ios7_osys_misc.mm

```diff
@@ -37,6 +45,74 @@ static inline void execute_on_main_thread_async(void (^block)(void)) {
 	}
 }
 
+void OSystem_iOS7::updateStartSettings(const Common::String &executable, Common::String &command, Common::StringMap &settings, Common::StringArray& additionalArgs) {
+		NSBundle* bundle = [NSBundle mainBundle];
+		// Check if scummvm is running from an app bundle
+		if (!bundle || ![bundle bundleIdentifier]) {
+			// Use default autostart implementation
+			EventsBaseBackend::updateStartSettings(executable, command, settings, additionalArgs);
+			return;
+		}
+
+		// If the bundle contains a scummvm.ini, use it as initial config
+		NSString *iniPath = [bundle pathForResource:@"scummvm" ofType:@"ini"];
+		if (iniPath && !settings.contains("initial-cfg"))
+			settings["initial-cfg"] = Common::String([iniPath fileSystemRepresentation]);
+
+		// If a command was specified on the command line, do not override it
+		if (!command.empty())
+			return;
+
+		// Check if we have an autorun file with additional arguments
+		NSString *autorunPath = [bundle pathForResource:@"scummvm-autorun" ofType:nil];
+		if (autorunPath) {
+			Common::File autorun;
+			Common::String line;
+			if (autorun.open(Common::FSNode([autorunPath fileSystemRepresentation]))) {
+				while (!autorun.eos()) {
+					line = autorun.readLine();
+					if (!line.empty() && line[0] != '#')
+						additionalArgs.push_back(line);
+				}
+			}
+			autorun.close();
+		}
+
+		// If the bundle contains a game directory, auto-detect it
+		NSString *gamePath = [[bundle resourcePath] stringByAppendingPathComponent:@"game"];
+		BOOL isDir = false;
+		BOOL exists = [[NSFileManager defaultManager] fileExistsAtPath:gamePath isDirectory:&isDir];
+		if (exists && isDir) {
+			// Ause auto-detection
+			command = "auto-detect";
+			settings["path"] = [gamePath fileSystemRepresentation];
+			return;
+		}
+
+		// The rest of the function has some commands executed only the first time after each version change
+		// Check the last version stored in the user settings.
+		NSString *versionString = [NSString stringWithUTF8String:gScummVMFullVersion];
+		NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+		NSString *lastVersion = [defaults stringForKey:@"lastVersion"];
+		if (lastVersion && [lastVersion isEqualToString:versionString])
+			return;
+		[defaults setObject:versionString forKey:@"lastVersion"];
+
+		// If the bundle contains a games directory, add them to the launcher
+		NSString *gamesPath = [[bundle resourcePath] stringByAppendingPathComponent:@"games"];
```

Comment:
Will this detection run every time at the scummvm startup?

Reply:
No. Commit 5a6fbc9 ensures this is only run the first time after the ScummVM version changes (so on first install and each subsequent upgrade).